File: academic-system/src/main/java/org/example/academic/system/service/ClassService.java
package org.example.academic.system.service;

import org.example.academic.system.AcademicSystem;
import org.example.academic.system.exception.AcademicSystemException;
import org.example.academic.system.model.AcademicClass;

public class ClassService {

    private AcademicSystem academicSystem;

    public ClassService(AcademicSystem academicSystem) {
        this.academicSystem = academicSystem;
    }

    public void registerClass(String code, String title) {
        if (code == null || code.isBlank()) {
            throw new AcademicSystemException("Class code cannot be blank.");
        }
        if (title == null || title.isBlank()) {
            throw new AcademicSystemException("Class title cannot be blank.");
        }
        academicSystem.addClass(new AcademicClass(code, title));
    }

    public void listClasses() {
        if (academicSystem.getClasses().isEmpty()) {
            System.out.println("No classes registered.");
            return;
        }
        System.out.println("\n--- Registered Classes ---");
        for (AcademicClass c : academicSystem.getClasses()) {
            System.out.println(c.getCode() + " | " + c.getTitle()
                    + " | Assessments: " + c.getAssessments().size());
        }
    }
}

File: academic-system-python/src/academic_system/service/class_service.py
from academic_system.academic_system import AcademicSystem
from academic_system.model.academic_class import AcademicClass
from academic_system.exception.exceptions import InvalidClassException
import logging

logger = logging.getLogger(__name__)


class ClassService:
    def __init__(self, system: AcademicSystem):
        self._system = system

    def register_class(self, code: str, title: str):
        if not code or not code.strip():
            raise InvalidClassException("Class code cannot be blank.")
        if not title or not title.strip():
            raise InvalidClassException("Class title cannot be blank.")
        self._system.add_class(AcademicClass(code.strip(), title.strip()))
        logger.info(f"Class registered: code={code}, title={title}")

    def list_classes(self):
        classes = self._system.get_classes()
        if not classes:
            print("No classes registered.")
            return
        print("\n--- Registered Classes ---")
        for c in classes:
            print(f"{c.code} | {c.title} | "
                  f"Assessments: {len(c.assessments)}")